### ISSUE TITLE
Replacing RxJava with Coroutines?

### ISSUE THREAD

**Issue by Petikoch** [sgrekov/Teapot#1]
Hi Sergey,

thank you very much for your blog posts (https://proandroiddev.com/taming-state-in-android-with-elm-architecture-and-kotlin-part-1-566caae0f706, ...)!

And thank you for sharing this library.

One question: 
Did you consider to write Teapot itself using Coroutines (Channel, Flow, ...) instead of RxJava? It's 2020 and Kotlin Coroutines get more and more traction and are "better integrated" in the Kotlin language than RxJava. 
And since Coroutines also feature "multi-platform" power, Teapot could also become multi-platform ready and we could write e.g. web-applications with it :-) 

Thanks for sharing your thoughts and best regards,
Peti

**Comment by sgrekov** [comment]
Hi Peti!
Thank you for kind words. Actually Teapot already have support of Coroutines(for executing effects) and of Flow(for doing subscriptions), you can even find it's usage in the sample ;) 
https://github.com/sgrekov/Teapot/blob/master/sample/src/main/java/dev/teapot/sample/main/feature/MainFeature.kt

But you are right, this is totally miss from docs perspective, there is no mention of it in the readme(

**Comment by Petikoch** [comment]
No, no... I saw that there is support for Coroutines. 

My question is why **it is built using RxJava** instead of **built using Coroutines**?

**Comment by sgrekov** [comment]
Ah, now I got what you mean) 
It **was** built with RxJava initially, because Coroutines weren't exist at the moment. But now it is actually **agnostic to effect library**. It doesn't require nor RxJava, nor Coroutines to work, if you will provide your own implementation of CommandExecutor.  In near future I plan to add support for more effect libraries, like Reaktive and ArrowFx. 
And also I already started to work on making this lib to be multiplatform (there is branch for it), and making it implementation Rx free was one of the step in that direction.
